cmd/aperturectl/cmd/blueprints.go

```diff
@@ -0,0 +1,144 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/jsonnet-bundler/jsonnet-bundler/pkg"
+	"github.com/jsonnet-bundler/jsonnet-bundler/pkg/jsonnetfile"
+	specv1 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v1"
+	"github.com/jsonnet-bundler/jsonnet-bundler/spec/v1/deps"
+	"github.com/spf13/cobra"
+)
+
+const apertureBlueprintsRepo = "github.com/fluxninja/aperture/blueprints"
+
+var (
+	blueprintsDir string
+
+	// Args for `blueprints pull`.
+	apertureVersion string
+)
+
+func init() {
+	rootCmd.AddCommand(blueprintsCmd)
+
+	blueprintsPullCmd.Flags().StringVar(&apertureVersion, "version", "main", "version of aperture to pull blueprints from")
+	blueprintsCmd.AddCommand(blueprintsPullCmd)
+}
+
+var blueprintsCmd = &cobra.Command{
+	Use:   "blueprints",
+	Short: "Manage blueprints",
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		userHomeDir, err := os.UserHomeDir()
+		if err != nil {
+			return err
+		}
+		blueprintsDir = filepath.Join(userHomeDir, ".aperturectl", "blueprints")
+		err = os.MkdirAll(blueprintsDir, os.ModePerm)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+var blueprintsPullCmd = &cobra.Command{
+	Use:   "pull",
+	Short: "Pull a blueprint",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		apertureBlueprintsDir := filepath.Join(blueprintsDir, apertureVersion)
+		err := os.MkdirAll(apertureBlueprintsDir, os.ModePerm)
+		if err != nil {
+			return err
+		}
+
+		spec := specv1.New()
+		contents, err := json.MarshalIndent(spec, "", "  ")
+		if err != nil {
+			return err
+		}
+		spec.LegacyImports = false
+		contents = append(contents, []byte("\n")...)
+
+		filename := filepath.Join(apertureBlueprintsDir, jsonnetfile.File)
+		err = os.WriteFile(filename, contents, os.ModePerm)
+		if err != nil {
+			return err
+		}
+
+		jbLockFileBytes, err := os.ReadFile(filepath.Join(apertureBlueprintsDir, jsonnetfile.LockFile))
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		lockFile, err := jsonnetfile.Unmarshal(jbLockFileBytes)
+		if err != nil {
+			return err
+		}
+
+		err = os.MkdirAll(filepath.Join(apertureBlueprintsDir, ".tmp"), os.ModePerm)
+		if err != nil {
+			return err
+		}
+
+		uri := fmt.Sprintf("%s@%s", apertureBlueprintsRepo, apertureVersion)
```

Comment:
we should also allow users to override the default URI -- in case they want to use local directory or their fork of our blueprints repo

Reply:
[23dce17](https://github.com/fluxninja/aperture/pull/1217/commits/23dce1754d56e27a8f965953f2fae8f2d31133cf) addresses this.